keps/sig-storage/5489-default-storage-class-per-access-mode/README.md

```diff
@@ -0,0 +1,907 @@
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+The primary drawback of this proposal is the introduction of additional configuration complexity for cluster administrators.
+
+* **Increased Mental Overhead:** Instead of a single, unambiguous global default `StorageClass`, administrators now have a more complex system to manage. They must be aware of mode-specific defaults, the global fallback, and the priority order used for tie-breaking. This could lead to misconfigurations where the resulting default `StorageClass` is not what was intended.
+
+* **Non-Obvious Priority for Multiple Access Modes:** For a `PersistentVolumeClaim` that requests multiple access modes (e.g., `["ReadWriteOnce", "ReadWriteMany"]`), the defaulting logic relies on a fixed priority hierarchy (`ReadWriteMany` > `ReadOnlyMany` > `ReadWriteOnce`). This behavior, while deterministic, might not be immediately obvious to users, who might expect the first mode listed in the array to take precedence. This could lead to volumes being provisioned from an unexpected `StorageClass`.
+
+## Alternatives
+
+Several out-of-tree and alternative in-tree solutions were considered. While flexible, they introduce significant challenges related to ordering and compatibility with existing Kubernetes components, making them less ideal than the proposed annotation-based approach.
+
+---
+
+#### Standard Field on StorageClass
+
+This approach would add a new field (e.g., `defaultForAccessMode`) directly to the `StorageClass` API, making the defaulting mechanism a first-class, versioned part of Kubernetes.
+
+* **Pros:**
+    * **Officially Supported and Discoverable:** The field would be part of the OpenAPI schema, making it visible and documented for users and tools. This improves discoverability and clarity.
+    * **Easier for Tooling and Validation:** The API server could validate that only one StorageClass is default per access mode, reducing misconfiguration. Tools and IDEs could provide better support for official fields.
+
+* **Cons:**
+    * **Requires an API Change:** Adding a new field to a stable API requires a full review, versioning, and careful backward compatibility considerations. This process is slow and has a high bar for acceptance.
+    * **Slower Iteration:** Changes to core APIs take longer to design, approve, and roll out. Any future improvements or fixes would also be slow.
+    * **Upgrade/Downgrade Complexity:** Introducing new fields can complicate cluster upgrades and downgrades, especially if components run different versions.
```

Comment:
annotations introduce a unique problem of field-squatting, where the annotations have values, but are not honored and then suddenly on an upgrade become active.